microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/core/java/diagnostics/JavaDiagnosticsContext.java

```diff
@@ -31,15 +34,33 @@
 
 	private final DocumentFormat documentFormat;
 
-	public JavaDiagnosticsContext(String uri, ITypeRoot typeRoot, IJDTUtils utils, DocumentFormat documentFormat) {
+	private final MicroProfileJavaDiagnosticsSettings settings;
+
+	public JavaDiagnosticsContext(String uri, ITypeRoot typeRoot, IJDTUtils utils, DocumentFormat documentFormat, MicroProfileJavaDiagnosticsSettings settings) {
 		super(uri, typeRoot, utils);
 		this.documentFormat = documentFormat;
+		if (settings == null) {
+			this.settings = new MicroProfileJavaDiagnosticsSettings(Collections.emptyList());
+		} else {
+			this.settings = settings;
+		}
 	}
 
 	public DocumentFormat getDocumentFormat() {
 		return documentFormat;
 	}
 
+	/**
+	 * Returns the MicroProfileJavaDiagnosticsSettings.
+	 *
+	 * Should not be null.
+	 *
+	 * @return the MicroProfileJavaDiagnosticsSettings
+	 */
+	public MicroProfileJavaDiagnosticsSettings getSettings() {

```

Comment:
Let's keep like this, but this settings should be store on JDT LS side as singleton.

Reply:
We talked about making a SharedSettings for the jdt-ls side. Is this what you mean?

Reply:
Yes exactly

Reply:
https://github.com/eclipse/lsp4mp/issues/163